paddle/framework/init.cc

```diff
@@ -48,13 +49,17 @@ void InitDevices() {
   places.emplace_back(platform::CPUPlace());
 
 #ifdef PADDLE_WITH_CUDA
-  int count = platform::GetCUDADeviceCount();
-  for (int i = 0; i < count; ++i) {
-    places.emplace_back(platform::CUDAPlace(i));
+  try {
+    int count = platform::GetCUDADeviceCount();
+    for (int i = 0; i < count; ++i) {
+      places.emplace_back(platform::CUDAPlace(i));
+    }
+  } catch (const std::exception &exp) {
+    LOG(WARNING) << "Compiled with WITH_CUDA, but no CUDA found in runtime.";
   }
 #else
   LOG(WARNING)
-      << "'GPU' is not supported, Please re-compile with WITH_GPU option";
+      << "'CUDA' is not supported, Please re-compile with WITH_CUDA option";
```

Comment:
Currently, `CMakeLists.txt` does not have `WITH_CUDA` option.

Reply:
done